core/fetcher.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	rpctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	Client
+
+	newBlockCh chan *block.Raw
+}
+
+// NewBlockFetcher returns a new `BlockFetcher`.
+func NewBlockFetcher(client Client) *BlockFetcher {
+	return &BlockFetcher{
+		Client: client,
+	}
+}
+
+// GetBlock queries Core for a `Block` at the given height.
+func (f *BlockFetcher) GetBlock(ctx context.Context, height *int64) (*block.Raw, error) {
+	raw, err := f.Block(ctx, height)
+	return raw.Block, err
+}
+
+// SubscribeNewBlockEvent subscribes to new block events from Core, returning
+// a new block event channel on success.
+func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan *block.Raw, error) {
+	// start the client if not started yet
+	if !f.IsRunning() {
+		if err := f.Start(); err != nil {
+			return nil, err
+		}
+	}
+	eventChan, err := f.Subscribe(ctx, newBlockSubscriber, newBlockEventQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	// create a wrapper channel for translating ResultEvent to "raw" block
+	if f.newBlockCh != nil {
+		return nil, fmt.Errorf("new block event channel exists")
+	}
+	newBlockChan := make(chan *block.Raw)
+	f.newBlockCh = newBlockChan
```

Comment:
This likely should be protected with a lock in case the BlockFetcher is meant to be used concurrently or documented that this structure is not thread-safe. Note that the fact that we have only one instance per app of BlockFecther implies that it can be used in multiple places, thus routines.

Reply:
Yes I had that there to begin with, but then @mattdf told me it's probably not necessary.